mitodl/micromasters
# New course enrollment UX

#### What are the relevant tickets?
Fixes #2467.

#### What's this PR do?
Changes the user experience for enrolling in courses through MicroMasters. Adds a message above the course listing in a program, explaining what the user's price is for these courses. Changes the buttons to say "Enroll Now" instead of displaying the course price. When the user clicks one of these "Enroll Now" buttons, the user sees a modal with two options: pay for the course, or audit it for free.

#### How should this be manually tested?
1. While enrolled in a program that has some courses that you are not yet enrolled in, visit the dashboard. Verify that the buttons to the right of the courses say "Enroll Now", and do not have any course price in the button text. Verify that there is no "Enroll and pay later" link. Click on the button, and verify that a dialog pops up with two choices: "Pay Now" and "Audit for Free & Pay Later". When clicked, the "Pay Now" button should redirect the user to the order summary page. When clicked, the "Audit" button should cause the user to be enrolled in the course on edX on the audit track.
2. Courses where you are already enrolled as an auditing user should instead have a button that says "Pay Now". Clicking on this button should not open a dialog, but should immediately redirect the user to the order summary page.
3. For a course without financial aid available, verify that the cost of the course is described in the blue box at the top of the course list. When the course enrollment dialog is opened, verify that the "Pay Now" button also includes the cost of the course.
4. Create a coupon for this course, and apply the coupon to your user account. Verify that the reduced price is correctly reflected in the text of the blue box above the course list, and in the "Pay Now" button.
5. For a course _with_ financial aid available, verify that the "Pay Now" button in the enrollment dialog is disabled before the user has calculated their financial aid. The text of the blue box should also direct the user to calculate their financial aid. Verify that the user can still audit the course.
6. After financial aid has been calculated, verify that the blue box and the "Pay Now" button should display the cost with financial aid applied. The "Pay Now" button should no longer be disabled.
7. For a course with both financial aid and a coupon, the price in the blue box and the "Pay Now" button should reflect both discounts.
8. Create a 100% off coupon for a program. Verify that clicking on the "Enroll Now" button on the dashboard for a course in that program does _not_ cause the enrollment dialog to appear, but instead redirects the user to the order summary page immediately.

#### Screenshots (if appropriate)
<img width="643" alt="course-price-message" src="https://cloud.githubusercontent.com/assets/132355/23184859/5abd23f8-f84f-11e6-9226-a5847bc43dc8.png">
<img width="585" alt="enroll dialog disabled" src="https://cloud.githubusercontent.com/assets/132355/23184875/66eec08c-f84f-11e6-9c25-6d44592bd280.png">
<img width="772" alt="enroll dialog enabled" src="https://cloud.githubusercontent.com/assets/132355/23184878/69e4a0cc-f84f-11e6-8d5a-f8e537e99be5.png">

![audit](https://cloud.githubusercontent.com/assets/132355/23221740/dd74c63a-f8f3-11e6-9a23-e65e3fee4131.png)
